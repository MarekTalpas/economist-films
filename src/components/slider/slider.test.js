// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Slider from './slider';

const dummyData = [
  {
    title: 'Episode 1 Title',
    type: 'Episode 1',
  },
  {
    title: 'Episode 2 Title',
    type: 'Episode 2',
    thumbnail: {
      thumb: {
        url: 'xxx',
      },
    },
  },
  {
    title: 'Some title',
    type: 'Series',
  },
];

jest.mock('../slider-item/slider-item', () =>
  jest.fn(() => <div>SliderItem</div>),
);

test('Slider renders correctly', () => {
  const tree : string = renderer.create(<Slider data={dummyData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
