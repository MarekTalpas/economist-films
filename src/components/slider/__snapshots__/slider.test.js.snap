// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders correctly 1`] = `
<div
  className="slider"
>
  <div
    className="slider-title"
  >
    Slider Title
  </div>
  <div
    className="slider-items"
  >
    <div>
      SliderItem
    </div>
    <div>
      SliderItem
    </div>
    <div>
      SliderItem
    </div>
  </div>
</div>
`;
